File: PATTERNS/Pattern50/Pattern50.go
package main

import "fmt"

func main() {
	var n int
	num := 2
	fmt.Printf("Enter the number:")
	fmt.Scanf("%d", &n)

	for i := 1; i < 2*n; i++ {
		if i == 1 || i == n || i == 2*n-1 {
			for j := 1; j <= n; j++ {
				fmt.Printf("%d ", num)
				num += 2
			}
		} else {
			fmt.Printf("%d", num)
			num += 2
		}
		fmt.Println()
	}
}

File: PATTERNS/Pattern50/Pattern50.c
#include<stdio.h>
int main() 
    {
        int n;
        printf("Enter the number of rows:");
        scanf("\n%d",&n);
        int num = 2;
        for (int i = 0; i < 2 * n - 1; i++) {
            if (i % (n - 1) == 0) {
                for (int j = 0; j < n; j++) {
                    printf("%d ",num);
                    num += 2;
                }
                printf("\n");
            } else {
                printf("%d\n",num);
                num += 2;
            }
        }
    }